crates/db_views/post/src/impls.rs

```diff
@@ -344,8 +362,21 @@ impl<'a> PostQuery<'a> {
         .filter(post::scheduled_publish_time.is_null());
     }
 
-    if let Some(community_id) = o.community_id.or(largest_subscribed_for_prefetch) {
-      query = query.filter(post::community_id.eq(community_id));
+    match o.community_id {
+      Some(CommunityOrMulti::CommunityId(id)) => {
+        query = query.filter(post::community_id.eq(id));
+      }
+      Some(CommunityOrMulti::MultiCommunityId(id)) => {
+        let multi = MultiCommunity::read(pool, ReadParams::Id(id)).await?;
+        query = query.filter(post::community_id.eq_any(multi.entries));
```

Comment:
```suggestion
        let communities = multi_community_entry::table.filter(multi_community_entry::multi_community_id.eq(id));
        query = query.filter(post::community_id.eq_any(communities));
```

This subquery can be evaluated once for the whole query, so it shouldn't be a performance problem, and is better than a separate query

https://www.postgresql.org/docs/17/using-explain.html#:~:text=In%20the%20example%20above,operator%20be%20amenable%20to%20hashing

Reply:
This is not compiling and throws a cryptic diesel error:
```

error[E0277]: the trait bound `SelectStatement<FromClause<lemmy_db_schema_file::schema::multi_community_entry::table>, query_builder::select_clause::DefaultSelectClause<FromClause<lemmy_db_schema_file::schema::multi_community_entry::table>>, query_builder::distinct_clause::NoDistinctClause, query_builder::where_clause::WhereClause<diesel::expression::grouped::Grouped<diesel::expression::operators::Eq<multi_community_id, diesel::expression::bound::Bound<Integer, MultiCommunityId>>>>>: AsInExpression<Integer>` is not satisfied
--> crates/db_views/post/src/impls.rs:380:56
|
380 |         query = query.filter(post::community_id.eq_any(communities));
|                                                 ------ ^^^^^^^^^^^ the trait `AsInExpression<Integer>` is not implemented for `SelectStatement<FromClause<table>, DefaultSelectClause<FromClause<table>>, NoDistinctClause, WhereClause<Grouped<...>>>`
|                                                 |
|                                                 required by a bound introduced by this call
|
```

Maybe its missing the join to multi_community_entry table. I wasnt able to get the join working inside this match, so it would probably have to be done in the general `joins()` function. That seems like a bad idea because frontpage queries would run into the join limit.

Reply:
Just add `.select(multi_community_entry::community_id)`

Reply:
Ah didnt know that this works.